python/lsst/daf/butler/core/serverModels.py

```diff
@@ -0,0 +1,107 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = (
+    "QueryDatasetsModel",
+    "ExpressionQueryParameter",
+)
+
+"""Models used for client/server communication."""
+
+from typing import (
+    Any,
+    List,
+    Mapping,
+    Optional,
+    Union,
+)
+
+import re
+
+from pydantic import BaseModel
+
+from .utils import iterable, globToRegex
+
+# DataId and kwargs can only take limited types
+DataIdValues = Union[str, int, float]
+DataId = Mapping[str, DataIdValues]
```

Comment:
There are already some very similar (but not quite identical) type aliases at

https://github.com/lsst/daf_butler/blob/f94a57823ca904996b74b8bab85282891f2d18a5/python/lsst/daf/butler/core/dimensions/_coordinate.py#L62

and

https://github.com/lsst/daf_butler/blob/f94a57823ca904996b74b8bab85282891f2d18a5/python/lsst/daf/butler/core/dimensions/_coordinate.py#L685

Reply:
Yes, I saw those and I think I changed this code over the weekend. I thought I had pushed before I sent out the review request but clearly I hadn't. I did change it to drop the `float` but should have added the `None`.

It's funny that both those definitions for DataId value are in the same file.